Add `format` for trace ID propagation. (#5803)
apollographql/router: .changesets/feat_propagation_format.md

### Add `format` for trace ID propagation. ([PR #5803](https://github.com/apollographql/router/pull/5803))

> When propagating trace ID to subgraph via header is is now possible to specify the format.

Reviewer: ```suggestion
The router now supports specifying the format of trace IDs that are propagated to subgraphs via headers.

You can configure the format with the `format` option:
```

```yaml
telemetry:
  exporters:
    tracing: 
      propagation: 
        request: 
          header_name: "my_header"

>           format: uuid

Reviewer: ```suggestion
          # Must be in UUID form, with or without dashes
          format: uuid
```

```

> Note that incoming requests must some form of UUID either with or without dashes.

Reviewer: ```suggestion
Note that incoming requests must be some form of UUID, either with or without dashes.

To learn about supported formats, go to [`request` configuration reference](https://apollographql.com/docs/router/configuration/telemetry/exporters/tracing/overview#request-configuration-reference) docs. 
```
